### ISSUE TITLE
Would like to suggest making the extension appear dark/light mode according to user preference on browser

### ISSUE THREAD

**Issue by silvererudite** [mozilla/multi-account-containers#1511]
### behavior
I believe the extension would be more accessible if we could implement dark/light mode automatically according to user preference on browser. For ex: I have my browser set to dark theme and when I install the Add-on it should appear in dark mode and vice versa.

## Note:
asking for attention of @groovecoder and @maxxcrawford. If this is a desirable feature according to you guys i would love to work on it as part of my Outreachy contribution.

**Comment by silvererudite** [comment]
@groovecoder may I work on this as part of my Outreachy contribution?

**Comment by maxxcrawford** [comment]
@silvererudite Assigned!

**Comment by silvererudite** [comment]
@maxxcrawford currently I'm being able to customize according to user preference on his/her local machine's OS. You know with CSS property  prefers-color-scheme: . however I wanted to detect the browser preference and make the addon automatically adjust to that which this property doesn't. 

Do you know if there's a way to detect what theme the browser is using?

**Comment by groovecoder** [comment]
You can use https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/API/theme/getCurrent to get the current theme, and that object might have enough to tell whether it's "dark" or "light"?

**Comment by silvererudite** [comment]
@groovecoder thanks for the resource but **what is the correct way to use the getCurrent() function**. Before changing the addon files i was trying to see if I can retrieve my browser theme info on a separate extension I made for testing.  But when I try to naively call the getCurrent() function it outputs nothing in the console. I also tried the example from the link's page and it also shows nothing. 

I'm attaching the ss of my test extension files for reference
![Screenshot (21)](https://user-images.githubusercontent.com/24827548/66254574-f51f6a00-e799-11e9-9c64-fa5ec07a3d7a.png)
![Screenshot (24)](https://user-images.githubusercontent.com/24827548/66254576-f81a5a80-e799-11e9-89c8-2060ad3a20da.png)

**Comment by maxxcrawford** [comment]
@silvererudite In the [Contain Facebook project](https://github.com/mozilla/contain-facebook), we set two different icons within the manifest to cover this. 

See file entry here: https://github.com/mozilla/contain-facebook/blob/master/src/manifest.json#L29-L40

**Comment by silvererudite** [comment]
@maxxcrawford and @groovecoder  I was able to fix that turns out there was some issue with content_scripts. I am almost done with the solve but I am stuck at a weird problem. Currently the theme-changing functionality works according to user preference on browser. However when the popup goes in dark mode, 
![Screenshot (1)](https://user-images.githubusercontent.com/24827548/66395326-ed113580-e9f8-11e9-9389-0c5b9245ed44.png)
the middle part is still unchanged as you can see from the ss. I went on to inspect the popup.html file and i'm guessing the html code for that part is as marked below
![Screenshot (2)_LI](https://user-images.githubusercontent.com/24827548/66395800-25fdda00-e9fa-11e9-890b-38f1a61278e9.jpg)

but no matter how much JS I write to select that part for changing it doesn't work, I'm attaching the js file for reference as well
![Screenshot (4)](https://user-images.githubusercontent.com/24827548/66396195-07e4a980-e9fb-11e9-81da-25008ff6ebdd.png)

If I am looking at the wrong place pls point me to the part of the code of the unchanged white area in the extension.

**Comment by yellowwoods12** [comment]
Hi @silvererudite, "userContext-wrapper" is the class name of the container tab elements that you are want to set theme for. I experimented by changing the background color by updating the CSS as you can see in the following screenshot.
![Screenshot from 2019-10-11 16-38-38](https://user-images.githubusercontent.com/31931348/66647235-fd6a2000-ec45-11e9-991f-05b1e9085230.png)

Try using this as a selector and then implement it. Hope this helps. Let me know if you still have issues.

**Comment by silvererudite** [comment]
hey @yellowwoods12 ...interestingly enough this selector doesn't seem to work when I'm trying to manipulate using JS. Here have a look..
```


function getStyle(themeInfo) {
  
  if (themeInfo.colors.toolbar) 
  {
  
    document.body.style.backgroundColor= themeInfo.colors.toolbar;
  
    document.getElementById('current-tab').style.color=themeInfo.colors.tab_background_text;
   
  const headers = document.getElementsByTagName('h3');
  for (var i = 0; i < headers.length; i++) {
  headers[i].style.color =themeInfo.colors.tab_background_text;
  } 

  const paras= document.getElementsByTagName('p');
  for (var i = 0; i < paras.length; i++) {
    paras[i].style.color =themeInfo.colors.tab_background_text;
    } 
   }
   document.getElementById('sort-containers-link').style.color=themeInfo.colors.tab_background_text;

   var legends1= document.getElementsByTagName('legend');
  for (var i = 0; i < legends1.length; i++) {
    legends1[i].style.color =themeInfo.colors.tab_background_text;
    } 

//this is the part of code to change the white areas in popup 

    const tabl= document.getElementsByClassName(".userContext-wrapper");
    // eslint-disable-next-line no-var
    for (var i = 0; i < tabl.length; i++) {
      tabl[i].style.backgroundColor ='black';
    } 

}
async function getCurrentThemeInfo() 
{
  var themeInfo = await browser.theme.getCurrent();
  
  getStyle(themeInfo);
}
getCurrentThemeInfo();

```
this is the JS file that I inserted at the end of popup.html file. The selector does work when changing using CSS but I don't quite understand what is wrong when I use JS.

**Comment by maxxcrawford** [comment]
@silvererudite Go ahead and [open a PR](https://help.github.com/en/articles/creating-a-pull-request) for this issue and we can review. Thanks!

**Comment by silvererudite** [comment]
yup @maxxcrawford and @groovecoder already did. PR #1521 . Eagerly waiting your review.

**Comment by hackel** [comment]
I hope that this can be revived for the new UX in v7 at some point.  Following the system dark/light theme would be a most welcome addition.  It looks like @silvererudite put in a lot of good work on this.

**Comment by dannycolin** [comment]
Fixed by #1944 and #1964
